### PR TITLE
feat: Add proxy methods and `staking.payoutStakers`

### LINKED ISSUE #162: Add Proxy pallet calls
See https://github.com/paritytech/substrate/pull/6156/files

### PR DESCRIPTION
**Pull request by emostov**
closes #162 

- `proxy.addProxy`, `proxy.proxy`, `proxy.removeProxy`,`proxy.removeProxies`, `staking.payoutStakers` calls are added.
-  `@polkadot/api` has been bumped to `1.19.0-beta.1` in order to use the static metadata for unit tests.
- Deprecated metadata is added as a test utility in order to maintain unit tests of deprecated democracy methods.

I also added a `temp_new_method_test` folder within examples. I don’t intend for this folder to get merged, but rather they can be used by a reviewer to easily check that the methods work in an integrated environment. However, I do think its worth considering keeping the `proxy.proxy` example since I think it can be a little confusing to know what to pass as an argument for the call.

### CODE REVIEW

File: src/methods/index.ts
Comment (amaury1093) [inline src/methods/index.ts:3]: I would say to keep all calls inside `democracy`, some people might still be using it. A console.warn inside these functions should be enough.

**Comment by amaury1093** [comment]
re the failing test: payout_validator seems to have been removed, i'm not 100% sure with what it's been replaced with

File: src/methods/index.ts
Comment (joepetrowski) [reply src/methods/index.ts:3]: Yeah, these are still active on Kusama so we need them (although a runtime upgrade is in that will replace them). Also we don't want to remove `vote` from democracy, just the proxy stuff.

Comment (joepetrowski) [reply src/methods/index.ts:3]: Actually, I'm too slow for Kusama. That runtime upgrade already went through. So for Kusama/Polkadot, we just need `democracy.vote`, but agree we should leave the others in for other chains and mark them as deprecated.

File: src/methods/staking/payoutStakers.ts
Comment (joepetrowski) [inline src/methods/staking/payoutStakers.ts]:
```suggestion
   * The Stash account of a _validator._ Their nominators, up to, the maximum
   * `MaxNominatorRewardedPerValidator`, will also receive their rewards.
```

File: src/methods/staking/payoutStakers.ts
Comment (joepetrowski) [inline src/methods/staking/payoutStakers.ts]:
```suggestion
   * May be any era between `[current_era - history_depth; current_era]`. Substrate only
   * stores reward information for up to `history_depth` eras.
```

File: src/methods/staking/payoutStakers.ts
Comment (joepetrowski) [inline src/methods/staking/payoutStakers.ts]:
```suggestion
 * Any account can call this function, even if it is not one of the stakers.
```

File: src/methods/staking/payoutStakers.ts
Comment (joepetrowski) [inline src/methods/staking/payoutStakers.ts]:
```suggestion
 * This can only be called when `EraElectionStatus` is `Closed`.
```

File: src/methods/proxy/addProxy.ts
Comment (joepetrowski) [inline src/methods/proxy/addProxy.ts]: No, not here. This is _runtime_ specific and not _Substrate_ specific.

File: src/methods/proxy/addProxy.ts
Comment (joepetrowski) [inline src/methods/proxy/addProxy.ts]:
```suggestion
   * The account that the `caller` would like to make a proxy.
```

File: src/methods/proxy/addProxy.ts
Comment (joepetrowski) [inline src/methods/proxy/addProxy.ts]:
```suggestion
   * The permissions allowed for this proxy account. Available types are declared in the runtime.
```

**Review by joepetrowski — COMMENTED** [review]
haven't looked at everything yet

File: src/methods/index.ts
Comment (emostov) [reply src/methods/index.ts:3]: I put back in all democracy methods, but had to put in deprecated metadata in order to keep the tests passing.

Comment (amaury1093) [reply src/methods/index.ts:3]: Let's add a console.warn in the deprecated functions too. Even if other chains are still using them, it'll be a gentle reminder for them to update to latest substrate

Comment (amaury1093) [reply src/methods/index.ts:3]: > but had to put in deprecated metadata in order to keep the tests passing

In order to keep this repo easier to maintain, I would also be fine with removing tests for deprecated methods. @emostov I'll let you decide on this one.

File: src/util/testUtil.ts
Comment (amaury1093) [inline src/util/testUtil.ts:70]: Good idea to give explicit names to objects. Could you rename the old `TEST_OPTIONS` to `KUSAMA_TEST_OPTIONS`?

File: src/methods/staking/index.ts
Comment (amaury1093) [inline src/methods/staking/index.ts:6]: tbh I'd also keep these 2, just in case, and add a deprecation warning

**Review by amaury1093 — COMMENTED** [review]
lgtm overall, just some final small comments. Also, could you run `yarn docs`? This will update all the auto-generated docs

File: src/methods/democracy/closeProxy.ts
Comment (joepetrowski) [inline src/methods/democracy/closeProxy.ts:21]: @amaurymartiny should these warnings be in the docs? Either way is fine with me.

File: src/methods/proxy/proxy.ts
Comment (joepetrowski) [inline src/methods/proxy/proxy.ts]:
```suggestion
 * Dispatch the given `call` from an account for which the sender is authorized.
```

File: src/methods/proxy/proxy.ts
Comment (joepetrowski) [inline src/methods/proxy/proxy.ts]:
```suggestion
   * The account for which the sender is a proxy.
```

File: src/methods/proxy/removeProxies.ts
Comment (joepetrowski) [inline src/methods/proxy/removeProxies.ts]:
```suggestion
 * Unregister all proxy accounts for the sender.
```

File: src/methods/proxy/removeProxy.ts
Comment (joepetrowski) [inline src/methods/proxy/removeProxy.ts]:
```suggestion
   * The account that the `caller` would like to unregister.
```

File: src/methods/proxy/removeProxy.ts
Comment (joepetrowski) [inline src/methods/proxy/removeProxy.ts]:
```suggestion
   * The permissions currently enabled for the target proxy account.
```

File: src/methods/staking/payoutStakers.ts
Comment (joepetrowski) [inline src/methods/staking/payoutStakers.ts]:
```suggestion
   * `MaxNominatorRewardedPerValidator`, will also receive their rewards.
```

File: src/methods/staking/payoutStakers.ts
Comment (joepetrowski) [inline src/methods/staking/payoutStakers.ts]:
```suggestion
   * May be any era between `[current_era - history_depth; current_era]`. Substrate only.
   * Retains up to `history_depth` eras of reward information.
   */
```

**Review by joepetrowski — COMMENTED** [review]
Just some docs issues, code looks OK though

File: src/methods/democracy/closeProxy.ts
Comment (amaury1093) [reply src/methods/democracy/closeProxy.ts:21]: We can add a `@deprecated` in the doc comments, these functions will then show as deprecated in the generated docs.

File: package.json
Comment (joepetrowski) [inline package.json]:
```suggestion
    "@polkadot/api": "^1.19.0-beta.3",
```

File: src/methods/proxy/proxy.ts
Comment (joepetrowski) [reply src/methods/proxy/proxy.ts]: this still needs a change

File: src/methods/proxy/removeProxies.ts
Comment (joepetrowski) [reply src/methods/proxy/removeProxies.ts]: same

File: src/methods/proxy/removeProxy.ts
Comment (joepetrowski) [reply src/methods/proxy/removeProxy.ts]: same

File: src/methods/staking/payoutStakers.ts
Comment (joepetrowski) [inline src/methods/staking/payoutStakers.ts]:
```suggestion
   * May be any era between `[current_era - history_depth; current_era]`. Substrate only
   * retains up to `history_depth` eras of reward information.
```

File: src/util/testUtil.ts
Comment (joepetrowski) [inline src/util/testUtil.ts]:
```suggestion
  address: '15oF4uVJwmo4TdGW7VfQxNLavjCXviqxT9S1MgbjMNHr6Sp5', // seed "//Alice"
```

File: src/util/testUtil.ts
Comment (joepetrowski) [inline src/util/testUtil.ts]:
```suggestion
      proxy: '14E5nqKAp3oAJcmzgZhUD2RcptBeUBScxKHgJKU4HPNcKVf3', // seed "//Bob",
```

File: src/util/testUtil.ts
Comment (joepetrowski) [inline src/util/testUtil.ts]:
```suggestion
      real: '14E5nqKAp3oAJcmzgZhUD2RcptBeUBScxKHgJKU4HPNcKVf3', // seed "//Bob",
```

**Review by joepetrowski — COMMENTED** [review]
Almost there!

**Comment by joepetrowski** [comment]
One more thing, I would remove all the temp examples. They will lead to a lot of maintenance for any changes. Obviously useful during the PR but don't think they need to go into master.

File: docs/globals.md
Comment (amaury1093) [inline docs/globals.md]: could you add a @ignore comment so that it doesn't appear in the docs?

File: src/util/metadataStatic_api_v1_17_2.ts
Comment (amaury1093) [inline src/util/metadataStatic_api_v1_17_2.ts:1]: For other people reading this file in the future, I would add a comment on this file, to let them know what's special about this metadata.

Also, metadata are uniquely defined by chainName + specVersion, so this file could be called: `metadataStatic_substrate_{n}.ts`

**Review by amaury1093 — APPROVED** [review]
code lgtm! Tests needs to pass before merging

File: src/util/metadataStatic_api_v1_17_2.ts
Comment (emostov) [reply src/util/metadataStatic_api_v1_17_2.ts:1]: See new comment at top of file.

**Pull request merged**
